pkg/register/protection.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2017 by the contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package register
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/go-github/github"
+	"github.com/heptiolabs/sign-off-checker/pkg/constants"
+)
+
+func hasBranchProtection(gh *github.Client, org string, repo *github.Repository) (bool, error) {
+	contexts, resp, err := gh.Repositories.ListRequiredStatusChecksContexts(
+		context.TODO(),
+		org,
+		repo.GetName(),
+		repo.GetDefaultBranch(),
+	)
+	if resp != nil && resp.StatusCode == 404 {
+		// 404 means no branch protection has been configured at all
+		return false, nil
+	}
+	if err != nil {
+		return false, fmt.Errorf("Error getting branch protection configuration for %q: %v", repo.GetFullName(), err)
+	}
+
+	// look for any required status check with our context
+	for _, context := range contexts {
+		if context == constants.SignOffCheckerContext {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// addBranchProtection adds the expected branch protection configuration to
+// a repository's default branch (usually "master"). This is less straightforward
+// because of the way it's intertwined with other branch protection settings.
+// The GH API forces us to get+modify+set which means this could race with other
+// concurrent modifications.
+func addBranchProtection(gh *github.Client, org string, repo *github.Repository) error {
+	existing, resp, err := gh.Repositories.GetBranchProtection(context.TODO(), org, repo.GetName(), repo.GetDefaultBranch())
+	if err != nil && resp != nil && resp.StatusCode != 404 {
```

Comment:
Is `resp` a http.Response object? Does resp.Body need to be closed here and everywhere?